File: Parser/src/ScheduleParser.java
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.net.URI;

public class ScheduleParser {
    public static void main(String[] args) {
        try {
            String endDate = "2025-03-23";
            String startDate = "2025-03-17";
            String groupID = "59774";

            HttpClient client = HttpClient.newHttpClient();
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create("https://urfu.ru/api/v2/schedule/groups/" + groupID + "/schedule?date_gte=" + startDate + "&date_lte=" + endDate))
                    .header("User-Agent", "Mozilla/5.0")
                    .GET()
                    .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            //Лабораторная работа 10, Архипов Н.А.©, 1.3.3 "Как прочитать файл JSON"
            JSONParser parser = new JSONParser();
            Object obj = parser.parse(response.body());
            JSONObject jsonObject = (JSONObject) obj;

            JSONArray jsonArray = (JSONArray) jsonObject.get("events");

            for (Object o : jsonArray){
                JSONObject event = (JSONObject) o;
                System.out.print("Предмет: " + event.get("title") + " " + event.get("date") + " "
                        + event.get("timeBegin") + " " + event.get("timeEnd") + " "
                        + event.get("teacherName") + "\n");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: Parser/Parser/ScheduleParser.cs
using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;

class ScheduleParser
{
    static async Task Main()
    {
        try
        {
            string endDate = "2025-03-23";
            string startDate = "2025-03-17";
            string groupID = "59774";

            string url = $"https://urfu.ru/api/v2/schedule/groups/{groupID}/schedule?date_gte={startDate}&date_lte={endDate}";

            using HttpClient client = new HttpClient();
            client.DefaultRequestHeaders.Add("User-Agent", "Mozilla/5.0");

            HttpResponseMessage response = await client.GetAsync(url);
            response.EnsureSuccessStatusCode();

            string responseBody = await response.Content.ReadAsStringAsync();

            // Десериализация JSON
            using JsonDocument doc = JsonDocument.Parse(responseBody);
            JsonElement root = doc.RootElement;

            if (root.TryGetProperty("events", out JsonElement events))
            {
                foreach (JsonElement eventElement in events.EnumerateArray())
                {
                    string title = eventElement.GetProperty("title").GetString() ?? "Нет данных";
                    string date = eventElement.GetProperty("date").GetString() ?? "Нет данных";
                    string timeBegin = eventElement.GetProperty("timeBegin").GetString() ?? "Нет данных";
                    string timeEnd = eventElement.GetProperty("timeEnd").GetString() ?? "Нет данных";
                    string teacherName = eventElement.GetProperty("teacherName").GetString() ?? "Нет данных";

                    Console.WriteLine($"Предмет: {title} {date} {timeBegin} {timeEnd} {teacherName}");
                }
            }
            else
            {
                Console.WriteLine("Поле 'events' отсутствует в JSON-ответе.");
            }
        }
        catch (HttpRequestException e)
        {
            Console.WriteLine($"Ошибка HTTP-запроса: {e.Message}");
        }
        catch (JsonException e)
        {
            Console.WriteLine($"Ошибка обработки JSON: {e.Message}");
        }
        catch (Exception e)
        {
            Console.WriteLine($"Произошла ошибка: {e.Message}");
        }
    }
}